### ISSUE TITLE
[BUG] Issues with multiplayer, Null Reference

### ISSUE THREAD

**Issue by AlexxSeven** [Goobwabber/BeatsaberOptimizer#1]
**Beatsaber Optimizer Version and Download Source**
v0.1.0 and I'm pretty sure at some point it was posted in #PC-Mods on the BSMG discord, but I can't find it there so either I'm losing it or that is where I found it but it was removed.

**Your Platform**
PC Steam V1.16.1 & 1.16.2 (it was happening on both.)

**Describe the bug**
Causes multiplayer to hang on song start/song load

**To Reproduce**
1. Go to 'Multiplayer/Online'.
2. Host or join a lobby.
3. Select a song and click play.
4. Game gets stuck at a black screen.

**Expected behavior**
Load into multiplayer session.

**Log**
```
[DEBUG @ 12:50:02 | SiraUtil] Installing: OptimizationInstaller (BeatsaberOptimizer)
[CRITICAL @ 12:50:02 | UnityEngine] ZenjectException: Unable to resolve 'PlayerTransforms'. Object graph:
[CRITICAL @ 12:50:02 | UnityEngine] Zenject.DiContainer.Resolve (Zenject.InjectContext context) (at <d0da33e76e734eb2a6363882a54b9841>:0)
[CRITICAL @ 12:50:02 | UnityEngine] Zenject.DiContainer.ResolveId (System.Type contractType, System.Object identifier) (at <d0da33e76e734eb2a6363882a54b9841>:0)
[CRITICAL @ 12:50:02 | UnityEngine] Zenject.DiContainer.Resolve (System.Type contractType) (at <d0da33e76e734eb2a6363882a54b9841>:0)
[CRITICAL @ 12:50:02 | UnityEngine] Zenject.DiContainer.Resolve[TContract] () (at <d0da33e76e734eb2a6363882a54b9841>:0)
[CRITICAL @ 12:50:02 | UnityEngine] BeatsaberOptimizer.Installers.OptimizationInstaller.InstallBindings () (at <deac3f5df545498aa96c2154e21b664b>:0)
[CRITICAL @ 12:50:02 | UnityEngine] Zenject.Context.InstallInstallers (System.Collections.Generic.List`1[T] normalInstallers, System.Collections.Generic.List`1[T] normalInstallerTypes, System.Collections.Generic.List`1[T] scriptableObjectInstallers, System.Collections.Generic.List`1[T] installers, System.Collections.Generic.List`1[T] installerPrefabs) (at <d0da33e76e734eb2a6363882a54b9841>:0)
[CRITICAL @ 12:50:02 | UnityEngine] (wrapper dynamic-method) Zenject.Context.Zenject.Context.InstallInstallers_Patch2(Zenject.Context)
[CRITICAL @ 12:50:02 | UnityEngine] (wrapper dynamic-method) Zenject.SceneContext.Zenject.SceneContext.InstallBindings_Patch1(Zenject.SceneContext,System.Collections.Generic.List`1<UnityEngine.MonoBehaviour>)
[CRITICAL @ 12:50:02 | UnityEngine] Zenject.SceneContext.Install () (at <d0da33e76e734eb2a6363882a54b9841>:0)
[CRITICAL @ 12:50:02 | UnityEngine] Zenject.SceneContext.RunInternal () (at <d0da33e76e734eb2a6363882a54b9841>:0)
[CRITICAL @ 12:50:02 | UnityEngine] Zenject.RunnableContext.Run () (at <d0da33e76e734eb2a6363882a54b9841>:0)
[CRITICAL @ 12:50:02 | UnityEngine] Zenject.RunnableContext.Initialize () (at <d0da33e76e734eb2a6363882a54b9841>:0)
[CRITICAL @ 12:50:02 | UnityEngine] Zenject.SceneContext.Awake () (at <d0da33e76e734eb2a6363882a54b9841>:0)
[CRITICAL @ 12:50:02 | UnityEngine] UnityEngine.GameObject:SetActive(Boolean)
[CRITICAL @ 12:50:02 | UnityEngine] UnityScenesHelper:SetActiveRootObjectsInScene(Scene, Boolean)
[CRITICAL @ 12:50:02 | UnityEngine] GameScenesManager:SetActiveRootObjectsInScenes(List`1, Boolean)
[CRITICAL @ 12:50:02 | UnityEngine] <ScenesTransitionCoroutine>d__36:MoveNext()
[CRITICAL @ 12:50:02 | UnityEngine] UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
[CRITICAL @ 12:50:02 | UnityEngine] NullReferenceException: Object reference not set to an instance of an object
[CRITICAL @ 12:50:02 | UnityEngine] VRRenderingParamsSetup.OnEnable () (at <f466e163166b4073b6dd391947ee7cac>:0)
[CRITICAL @ 12:50:02 | UnityEngine] UnityEngine.GameObject:SetActive(Boolean)
[CRITICAL @ 12:50:02 | UnityEngine] UnityScenesHelper:SetActiveRootObjectsInScene(Scene, Boolean)
[CRITICAL @ 12:50:02 | UnityEngine] GameScenesManager:SetActiveRootObjectsInScenes(List`1, Boolean)
[CRITICAL @ 12:50:02 | UnityEngine] <ScenesTransitionCoroutine>d__36:MoveNext()
[CRITICAL @ 12:50:02 | UnityEngine] UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
[CRITICAL @ 12:50:02 | UnityEngine] NullReferenceException: Object reference not set to an instance of an object
[CRITICAL @ 12:50:02 | UnityEngine] FadeInOnSceneTransitionFinished.OnEnable () (at <f466e163166b4073b6dd391947ee7cac>:0)
[CRITICAL @ 12:50:02 | UnityEngine] UnityEngine.GameObject:SetActive(Boolean)
[CRITICAL @ 12:50:02 | UnityEngine] UnityScenesHelper:SetActiveRootObjectsInScene(Scene, Boolean)
[CRITICAL @ 12:50:02 | UnityEngine] GameScenesManager:SetActiveRootObjectsInScenes(List`1, Boolean)
[CRITICAL @ 12:50:02 | UnityEngine] <ScenesTransitionCoroutine>d__36:MoveNext()
[CRITICAL @ 12:50:02 | UnityEngine] UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
[INFO @ 12:50:02 | Custom Avatars] [PlayerAvatarManager] Player height set to 1.684758 m
[INFO @ 12:50:02 | Custom Avatars] [SpawnedAvatar(Lumine)] Avatar resized with scale: 1.580503
[CRITICAL @ 12:50:02 | UnityEngine] NullReferenceException: Object reference not set to an instance of an object
[CRITICAL @ 12:50:02 | UnityEngine] MultiplayerConditionalActiveByLayout.Start () (at <ed5f1dff68954975944aebd4c110b7aa>:0)
```

**Additional context**
I went through multiple logs carefully to determine at what point the game was throwing a NullReferenceException to try and determine which mod/mods were causing it, disabled/removed several till i noticed the following part of the log above. after which once it gets to `NullReferenceException` it starts to just loop like that listing different parts of the multiplayer layout until the game is closed.

**Comment by AlexxSeven** [comment]
Addendum:
Forgot to attach beefy boi log that has the loop. lol

[2021.06.20.12.47.38.log](https://github.com/Goobwabber/BeatsaberOptimizer/files/6682886/2021.06.20.12.47.38.log)

**Comment by Goobwabber** [comment]
fixed in bb46fb7
